src/ext/ddtrace.c

```diff
@@ -540,34 +669,58 @@ static PHP_FUNCTION(dd_trace_internal_fn) {
 #endif
 }
 
-/* {{{ proto string dd_trace_generate_id() */
-static PHP_FUNCTION(dd_trace_generate_id) {
-    PHP5_UNUSED(return_value_used, this_ptr, return_value_ptr, ht TSRMLS_CC);
-    PHP7_UNUSED(execute_data);
-
-#if PHP_VERSION_ID >= 70200
-    RETURN_STR(dd_trace_generate_id());
-#else
-    char buf[20];
-    dd_trace_generate_id(buf);
 #if PHP_VERSION_ID >= 70000
-    RETURN_STRING(buf);
+#define RETURN_SPAN_ID(id_fn)                            \
+    do {                                                 \
+        char buf[20];                                    \
+        php_sprintf(buf, "%" PRIu64, (id_fn)(TSRMLS_C)); \
```

Comment:
Its fine in this instance since the id is of known max lenght. But my spidey senses do tingle when not using snprintf with set buffer length.
Also I think we could safely use vanilla (non php) `snprintf/sprintf` here, WDYT ?

Reply:
🕷 👍 😆